File: BackendAPI/BE.DAL/Entities/Supplier.cs
namespace BackendAPI.BE.DAL.Entities;
using BackendAPI.BE.BLL.Interfaces;
public class Supplier: IEntity
{
    public int SupplierId { get; set; }   
    public int WarehouseId { get; set; } // FK to Warehouse   

    public string Name { get; set; } = string.Empty;
    public string Contact { get; set; } = string.Empty;
    public string Email { get; set; } = string.Empty;
    public string Phone { get; set; } = string.Empty;
    public string Address { get; set; } = string.Empty;
    public string getKey() => SupplierId.ToString();

    // Navigation properties
    public ICollection<ProductSupplier> ProductSuppliers { get; set; } = new List<ProductSupplier>();
    public Warehouse Warehouse { get; set; } = null!;
}

File: Frontend/src/types/supplier.ts
export interface Supplier {
  id: number;
  name: string;
  contact: string;
  email: string;
  phone: string;
  address: string;
}

export type SupplierInput = Omit<Supplier, "id">;

export interface SupplierApiResponse {
  id?: number;
  supplierId?: number;
  warehouseId?: number;
  name?: string | null;
  contact?: string | null;
  email?: string | null;
  phone?: string | null;
  address?: string | null;
}
